TrainingApp/Controller/SettingsViewController.swift

```diff
@@ -14,78 +14,133 @@ final class SettingsViewController: UIViewController {
     @IBOutlet private var bigRectangle: UIImageView!
     @IBOutlet private var flightUpdatesLabel: UILabel!
     @IBOutlet private var executiveProgramLabel: UILabel!
     @IBOutlet private var discountDealsLabel: UILabel!
-    @IBOutlet private var flightUpdatesToggle: UIButton!
-    @IBOutlet private var executiveProgramToggle: UIButton!
-    @IBOutlet private var discountDealsToggle: UIButton!
+    @IBOutlet private var flightUpdatesSwitch: UISwitch!
+    @IBOutlet private var executiveProgramSwitch: UISwitch!
+    @IBOutlet private var discountDealsSwitch: UISwitch!
+    
+    let userDefaults = UserDefaults.standard
+    
+    let ON_OFF_KEY_THEME = R.string.localizable.onOffKeyTheme()
+    let THEME_KEY = R.string.localizable.themeKey()
+    let DARK_THEME = R.string.localizable.darkTheme()
+    let LIGHT_THEME = R.string.localizable.lightTheme()
+    
+    let ON_OFF_KEY_NOTIFICATION = R.string.localizable.onOffKeyNotfn()
+    let NOTIFICATION_KEY = R.string.localizable.notfnKey()
+    let NOTIFICATION_ON = R.string.localizable.notfnOn()
+    let NOTIFICATION_OFF = R.string.localizable.notfnOff()
+    
+    let ON_OFF_KEY_FLIGHT = R.string.localizable.onOffKeyFlight()
+    let FLIGHT_KEY = R.string.localizable.flightKey()
+    let FLIGHT_ON = R.string.localizable.flightOn()
+    let FLIGHT_OFF = R.string.localizable.flightOff()
+    
+    let ON_OFF_KEY_EXECUTIVE = R.string.localizable.onOffKeyExecutive()
+    let EXECUTIVE_KEY = R.string.localizable.executiveKey()
+    let EXECUTIVE_ON = R.string.localizable.executiveOn()
+    let EXECUTIVE_OFF = R.string.localizable.executiveOff()
+    
+    let ON_OFF_KEY_DISCOUNT = R.string.localizable.onOffKeyDiscount()
+    let DISCOUNT_KEY = R.string.localizable.discountKey()
+    let DISCOUNT_ON = R.string.localizable.discountOn()
+    let DISCOUNT_OFF = R.string.localizable.discountOff()
     
     // MARK: - Lifecycle
     
     override func viewDidLoad() {
         super.viewDidLoad()
         
-        initialButtonsSetup()
         setupTapGestureForViews()
-    }
-    
-    // MARK: - Actions
-    
-    @IBAction private func darkModeTogglePressed(_ sender: UIButton) {
-        if darkModeToggleButton.currentImage == R.image.button() {
-            darkModeIsOn()
-        }
-        else {
-            darkModeIsOff()
+        checkThemeSwitchState()
+        checkNotificationSwitchState()
+        print(FLIGHT_KEY)
+        print(EXECUTIVE_KEY)
+        updateTheme()
+        updateNotifications()
+
+        darkModeSwitch.setOnValueChangeListener {
+            if self.darkModeSwitch.isOn {
+                self.userDefaults.set(true, forKey: self.ON_OFF_KEY_THEME)
+                self.userDefaults.set(self.DARK_THEME, forKey: self.THEME_KEY)
+            }
+            else {
+                self.userDefaults.set(false, forKey: self.ON_OFF_KEY_THEME)
+                self.userDefaults.set(self.LIGHT_THEME, forKey: self.THEME_KEY)
+            }
+            self.updateTheme()
```

Comment:
Don't use 60 lines in viewDidLoad()
Create new functions for example setupDarkTheme() and call it in  viewDidLoad()

Reply:
Done 👍
I created "setupSwitchListeners()" to setup listeners for all switches.